Mathlib/RingTheory/TensorProduct/Basic.lean

```diff
@@ -948,6 +948,57 @@ theorem congr_trans [Algebra S C] [IsScalarTower R S C]
 
 theorem congr_symm (f : A ≃ₐ[S] B) (g : C ≃ₐ[R] D) : congr f.symm g.symm = (congr f g).symm := rfl
 
+variable (R A B C D)
+
+/-- Tensor product of algebras analogue of `mul_left_comm`. -/
+def leftComm : A ⊗[R] B ⊗[R] C ≃ₐ[R] B ⊗[R] A ⊗[R] C :=
+  let e₁ := (Algebra.TensorProduct.assoc R A B C).symm
+  let e₂ := congr (Algebra.TensorProduct.comm R A B) (1 : C ≃ₐ[R] C)
+  let e₃ := Algebra.TensorProduct.assoc R B A C
+  e₁.trans (e₂.trans e₃)
+
+variable {A B C D}
+
+@[simp]
+theorem leftComm_tmul (m : A) (n : B) (p : C) :
+    leftComm R A B C (m ⊗ₜ (n ⊗ₜ p)) = n ⊗ₜ (m ⊗ₜ p) :=
+  rfl
+
+@[simp]
+theorem leftComm_symm_tmul (m : A) (n : B) (p : C) :
+    (leftComm R A B C).symm (n ⊗ₜ (m ⊗ₜ p)) = m ⊗ₜ (n ⊗ₜ p) :=
+  rfl
+
+@[simp]
+theorem leftComm_toLinearEquiv :
+    (leftComm R A B C : _ ≃ₗ[R] _) = _root_.TensorProduct.leftComm R A B C := rfl
+
+variable (A B C D)
+
+/-- Tensor product of algebras analogue of `mul_mul_mul_comm`. -/
+def tensorTensorTensorComm : (A ⊗[R] B) ⊗[R] C ⊗[R] D ≃ₐ[R] (A ⊗[R] C) ⊗[R] B ⊗[R] D :=
+  let e₁ := Algebra.TensorProduct.assoc R A B (C ⊗[R] D)
+  let e₂ := congr (1 : A ≃ₐ[R] A) (leftComm R B C D)
+  let e₃ := (Algebra.TensorProduct.assoc R A C (B ⊗[R] D)).symm
+  e₁.trans (e₂.trans e₃)
+
+variable {M N P Q}
+
+@[simp]
+theorem tensorTensorTensorComm_tmul (m : A) (n : B) (p : C) (q : D) :
+    tensorTensorTensorComm R A B C D (m ⊗ₜ n ⊗ₜ (p ⊗ₜ q)) = m ⊗ₜ p ⊗ₜ (n ⊗ₜ q) :=
+  rfl
+
+@[simp]
+theorem tensorTensorTensorComm_symm :
+    (tensorTensorTensorComm R A B C D).symm = tensorTensorTensorComm R A C B D := by
+  ext; rfl
+
+@[simp]
+theorem tensorTensorTensorComm_toLinearEquiv :
+    (tensorTensorTensorComm R A B C D : _ ≃ₗ[R] _)
+      = _root_.TensorProduct.tensorTensorTensorComm R A B C D := rfl
```

Comment:
```suggestion
    (tensorTensorTensorComm R A B C D : _ ≃ₗ[R] _) =
      _root_.TensorProduct.tensorTensorTensorComm R A B C D := rfl
```

Reply:
Thanks for the review - I've applied both comments.